### PR TITLE
*: update kvproto.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @qiuyesuifeng @shenli @disksing  @huachaohuang

### CODE REVIEW

File: server/leader.go [hunk 1]
```diff
@@ -113,8 +113,8 @@ func (s *Server) isSameLeader(leader *pdpb.Leader) bool {
 
 func (s *Server) marshalLeader() string {
 	leader := &pdpb.Leader{
-		Addr: proto.String(s.GetAddr()),
-		Pid:  proto.Int64(int64(os.Getpid())),
+		Addr: s.GetAddr(),
+		Pid:  int64(os.Getpid()),
 	}
 
 	data, err := proto.Marshal(leader)
```
Comment (shenli) [inline server/leader.go]: You can use leader.Marshal() instead of proto.Marshal(). So as proto.Unmarshal().

**Comment by siddontang** [comment]
PTAL @shenli 
I will update tidb vendor after PD merged.

**Comment by shenli** [comment]
LGTM

File: server/leader.go (hunk 1 above)
Comment (qiuyesuifeng) [inline server/leader.go]: @siddontang how about this?

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**